### PR TITLE
chore(docs): cleanup authorize-push script

### PR DESCRIPTION
**Pull request by boennemann**
There wasn't a real problem, but I just generally cleaned it up and fixed one edge case, so this should be a little more future proof.

### CODE REVIEW

File: bin/authorise-push.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
 #!/usr/bin/env node
 
 var exec = require('child_process').exec
+var url = require('url')
 
 var GH_TOKEN = process.env.GH_TOKEN
-var repo = require('../package.json').repository.url
+var repo = url.parse(require('../package.json').repository.url)
+repo.auth = GH_TOKEN
```
Comment (boennemann) [inline bin/authorise-push.js]: core url module > RegExp fiddling

File: bin/authorise-push.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
-if (!(process.env.CI && GH_TOKEN && repo)) {
+if (!(process.env.CI === 'true' && GH_TOKEN && repo)) {
```
Comment (boennemann) [inline bin/authorise-push.js]: I got bitten by this once. The `CI` variable usually is the string "true" or "false", which means CI=false would pass this condition.

File: bin/authorise-push.js [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
   process.exit(1)
 }
 
-exec('git remote set-url origin ' + repo.replace('https://', 'https://' + GH_TOKEN + '@') + ' && ')
-exec('git config user.email gregor.martynus@ehealthnigeria.org && ')
+exec('git remote set-url origin ' + url.format(repo))
+exec('git config user.email gregor.martynus@ehealthnigeria.org')
```
Comment (boennemann) [inline bin/authorise-push.js]: FYI: You don't need to put a real email address here, because it's already authed by the token.

**Comment by gr2m** [comment]
thanks! Merged via aa70393

**Pull request closed without merging**